roachpb/batch.go

```diff
@@ -47,16 +47,19 @@ func (ba *BatchRequest) SetActiveTimestamp(nowFn func() hlc.Timestamp) error {
 
 // IsFreeze returns whether the batch consists of a single ChangeFrozen request.
 func (ba *BatchRequest) IsFreeze() bool {
-	if len(ba.Requests) != 1 {
+	if !ba.IsSingleRequest() {
 		return false
 	}
 	_, ok := ba.GetArg(ChangeFrozen)
 	return ok
 }
 
-// IsLease returns whether the batch consists of a single RequestLease request.
-func (ba *BatchRequest) IsLease() bool {
-	if len(ba.Requests) != 1 {
+// IsLeaseRequest returns whether the batch consists of a single RequestLease
+// request. Note that TransferLease requests return false.
+// RequestLease requests are special because they're the only type of requests a
+// non-lease-holder can propose.
```

Comment:
This description sounds like the `skipLeaseCheck` flag, but that flag is both on `RequestLease` and `TransferLease` (and that in turn is justified inline in the flags on `TransferLeaseRequest`). Some of this confusion should be reducible.